### PR TITLE
Upgrade KubernetesClient dependency

### PR DESCRIPTION
**Pull request by tomachristian**
Fixes https://github.com/microsoft/ApplicationInsights-Kubernetes/issues/357.

### CODE REVIEW

**Comment by didranoqx** [comment]
Hi, this commit is important for me. Have you some ideas when it could be approved ?

**Comment by tomachristian** [comment]
Hey @xiaomi7732, sorry to mention you directly, but do you think you could help us with this?

File: src/ApplicationInsights.Kubernetes/ApplicationInsights.Kubernetes.csproj [hunk 1]
```diff
@@ -18,7 +18,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="KubernetesClient" Version="[10.0.16, 11.0.0)" />
+    <PackageReference Include="KubernetesClient" Version="12.1.1" />
```
Comment (xiaomi7732) [inline src/ApplicationInsights.Kubernetes/ApplicationInsights.Kubernetes.csproj:21]: This bumps up the lowest version supported.
If I understand correct, the version needs to be uncapped to unblock the bumping up of the dependencies. How about make it `10.0.16`? That would be equivalent to `[10.0.16, )`.

Considerations:
1. Backward compatibility for existing code relying on the lower version;
2. Potentially breaking changes introduced in the K8sClient (major version upgrades);

If that would work for you, could you please update the PR?

**Review by xiaomi7732 — CHANGES_REQUESTED** [review]

File: src/ApplicationInsights.Kubernetes/ApplicationInsights.Kubernetes.csproj (hunk 1 above)
Comment (tomachristian) [reply src/ApplicationInsights.Kubernetes/ApplicationInsights.Kubernetes.csproj]: I upgraded the reference as per your request. Thank you!

**Review by xiaomi7732 — APPROVED** [review]
LGTM. Thanks for the contribution!

File: src/ApplicationInsights.Kubernetes/ApplicationInsights.Kubernetes.csproj (hunk 1 above)
Comment (xiaomi7732) [reply src/ApplicationInsights.Kubernetes/ApplicationInsights.Kubernetes.csproj]: Cool!

**Pull request merged**